libvast/vast/concept/parseable/string/quoted_string.hpp

```diff
@@ -22,22 +22,51 @@
 
 namespace vast {
 
+/// Parser for a string surrounded by a pair of quote characters `Quote`. Inside
+/// the  string, the escape sequence `Esc Quote` can be used to represent a
+/// literal quote, and the escape sequence `Esc Esc` can be used to represent
+/// two occurences of the escape character. All other occurences of the escape
+/// character are interpreted as character literals.
+///
+/// For example, using backslash as escape character and double quote as quote
+/// character:
+///
+///    "foo\n"     denotes the string ['f', 'o', 'o', '\', 'n']
+///    "C:\sys32"  denotes the string ['C', ':', '\', 's', 'y', 's', '3', '2']
+///    "\\"        denotes the string ['\', '\']
+///    "\""        denotes the string ['"']
+///    "\\""       is an invalid string (junk after end of string)
+///    "\\\"       is an invalid string (unterminated)
```

Comment:
`@note` can be used to highlight a Doxygen comment as a notice. Otherwise it will appear as the class description for `quoted_string_parser`, which doesn't really fit.